Dev guide: review read data (#8666)
pingcap/docs: develop/dev-guide-hybrid-oltp-and-olap-queries.md

1 row in set (0.07 sec)
```

> After replicas added, you can use the `EXPLAIN` statement to check out the execution plan of the above window function [partition by clause](#partition-by-clause). If `cop[tiflash]` appears in the execution plan, it means that the TiFlash engine has started to work.

Reviewer: ```suggestion
After replicas are added, you can use the `EXPLAIN` statement to check the execution plan of the above window function [`PARTITION BY` clause](#partition-by-clause). If `cop[tiflash]` appears in the execution plan, it means that the TiFlash engine has started to work.
```

> Execute [partition by clause](#partition-by-clause) again, the result is as follow:

Reviewer: grammar mistake

1500 rows in set (0.79 sec)
```

By comparing the two execution results, you can find that the query speed is significantly improved with TiFlash (the improvement is more significant with larger amount of data). This is because full table scan is needed when using window functions, and columnar TiFlash is better suited to handle this type of analytical task than row-based TiKV. For TiKV, using primary keys or indexed to reduce the number of rows, it will be very fast and consume fewer resources compared to TiFlash.

### Specify the query engine

TiDB uses the Cost Based Optimizer (CBO) to automatically choose whether to use TiFlash replicas based on cost estimates. However, if you are sure about the type of query in practice, you can specify the query engine to be used with [Optimizer Hints](/optimizer-hints.md).

You can use Hint `/*+ read_from_storage(engine_name[table_name]) */` to specify the query engine to be used in the following statement.:

> **Notice:**
>
> 1. If the table has an alias, use the alias in Hint, otherwise, Hint will not work.

> > 2. Additionally, setting read_from_storage Hint for [common table expression](/develop/dev-guide-use-common-table-expression.md) does not work.

Reviewer: ```suggestion
> 2. The `read_from_storage` hint does not work for [common table expression](/develop/dev-guide-use-common-table-expression.md).
```

SELECT * FROM acc;
```

> You can use the `EXPLAIN` statement to check out the execution plan of the above SQL. If `cop[tiflash]` and `cop[tikv]` appear in the task column at the same time, it means that TiFlash and TiKV are scheduled to complete this query. It is worthy to note that TiFlash and TiKV storage engines are usually part of different TiDB nodes, so the two query types are not affected by each other.

Reviewer: ```suggestion
You can use the `EXPLAIN` statement to check the execution plan of the above SQL statement. If `cop[tiflash]` and `cop[tikv]` appear in the task column at the same time, it means that TiFlash and TiKV are both scheduled to complete this query. Note that TiFlash and TiKV storage engines usually use different TiDB nodes, so the two query types are not affected by each other.
```
